Makefile.include

```diff
@@ -61,6 +61,8 @@ DLCACHE                         ?= $(RIOTTOOLS)/dlcache/dlcache.sh
 DLCACHE_DIR                     ?= $(RIOTBASE)/.dlcache
 RIOT_VERSION_DUMMY_CODE         ?= RIOT_VERSION_NUM\(2042,5,23,0\)
 
+CFLAGS :=
```

Comment:
Wouldn't this prevent from passing cflags from the command line? Would the following suggestion work instead?
```suggestion
CFLAGS ?=
```

Reply:
It would indeed prevent passing CFLAGS from the environment. But `?=` would also be a recursively expanded one (both when it comes from the environment and when it is set in the Makefile).

Note that some distros do actually set their default CFLAGS in some /etc/profile.d. Those flags might not be super ideal.

There are also ordering issues: Variables are assigned prior to their existence. Clearly, it would be better to just fix the ordering issue. Any `SUPPORTS_SOME_FLAG := $(shell $(CC) $(CFLAGS) -some-flag-to-test) && echo 1 || echo 0` done before the variables references becoming valid would fail.

A quick fix to just get the evaluation once per module fixed would be to just add `CFLAGS := $(CFLAGS)` after they are all populated. Also the subsequent `CFGLAGS += ...` we have in some Makefiles would still work afterwards and keep the type of the variable as simple expanded.